util/conditions/setter.go

```diff
@@ -96,6 +96,25 @@ func FalseCondition(t clusterv1.ConditionType, reason string, severity clusterv1
 	}
 }
 
+// NegativeTrueCondition returns a negative polarity condition with Status=True and the given type (Status=True has a negative meaning).
+func NegativeTrueCondition(t clusterv1.ConditionType, reason string, severity clusterv1.ConditionSeverity, messageFormat string, messageArgs ...interface{}) *clusterv1.Condition {
```

Comment:
why do NegativeTrueCondition and PositiveFalseCondition include the polarity in the func name?
Since all they are doing is set values in a clusterv1.Condition type wouldn't they be called just TrueCondition / falseCondition and let higher invoker decide on polarity?

Reply:
See https://github.com/kubernetes-sigs/cluster-api/pull/10550#discussion_r1588979694, but the intent is to keep things consistent with what we have.

Just a note about the approach you are suggesting about let higher invoker decide on polarity: this is not in discussion.

The difference is that:
- If we change the existing TrueCondition & FalseCondition, this will impact everyone and we end up to defer to users the task to enforce the same consistency that today we have with this helpers
- With the proposed change instead, the users specific helpers for the NegativePolarity, which are consistent with the existing one for positive polarity

Last, but not least, there is already a Set helper with allow the users to decide which field to set (but in this case, it also imply the task to enforce consistency across the codebase/all the providers)